index.js

```diff
@@ -1,68 +1,65 @@
+import { monsters } from './monsters.js';
 
-import {monsters} from './monsters.js';
-console.log(monsters);
-for(let monsterDiv of monsters){
-    showMonster(monsterDiv);
+const monsterContainer = document.querySelector('.monsters');
+const monsterFilter = document.querySelector('#search-monster');
 
+function populateMonsters(content) {
+  // Clear out existing content
+  monsterContainer.innerHTML = '';
+  // Usde showMonster to generate an Array of DOMNodes
+  let createMonsters = content.map((el) => showMonster(el));
+  // Append the DOMNodes in one DOM transaction
+  monsterContainer.append(...createMonsters);
 }
 
-function showMonster(monsterDiv){
-    const monster = document.createElement('div');
-    monster.className = 'monster';
-    const img= document.createElement('img');
-    img.src = `https://robohash.org/${monsterDiv.id}?set=set2`;
-    img.alt = monsterDiv.name ;
-    const mname = document.createElement('p');
-    mname.className = 'name' ;
-    mname.innerText = monsterDiv.name;
-    const email = document.createElement('p');
-    email.className = 'email'
-    email.innerText = monsterDiv.email;
-    monster.append(img,mname,email);
-    document.querySelector('.monsters').append(monster);
-    console.log(monster);
-}
+function showMonster(monsterDiv) {
+  const monster = document.createElement('div');
+  monster.className = 'monster';
 
+  const img = document.createElement('img');
+  img.src = `https://robohash.org/${monsterDiv.id}?set=set2`;
+  img.alt = monsterDiv.name;
 
-function showMonsterNotFound() {
-    const notFound= document.createElement('div');
-    notFound.className='p-5 not-found';
-    // notFound.style.display='none';
-    const span = document.createElement('span');
-    span.innerText='404';
-    const h1 = document.createElement('h1')
-    h1.innerText='🧟‍♂️ No Monster Found 🧟‍♂️' ;
-    notFound.append(span,h1);
-    document.querySelector('.monsters').append(notFound);
-    console.log(notFound);
+  const mname = document.createElement('p');
+  mname.className = 'name';
+  mname.innerText = monsterDiv.name;
 
+  const email = document.createElement('p');
+  email.className = 'email';
+  email.innerText = monsterDiv.email;
+  monster.append(img, mname, email);
 
+  return monster;
 }
-showMonsterNotFound()
 
-document.querySelector('#search-monster').addEventListener('keyup', function(e) {
-    const keyword= e.target.value.toLowerCase();
-    const findMonster = document.querySelectorAll('.monster');
-    let notFound = true;
+function showMonsterNotFound() {
+  const notFound = document.createElement('div');
+  notFound.className = 'p-5 not-found';
+  const span = document.createElement('span');
+  span.innerText = '404';
+  const h1 = document.createElement('h1');
+  h1.innerText = '🧟‍♂️ No Monster Found 🧟‍♂️';
+  notFound.append(span, h1);
+  return notFound;
+}
```

Comment:
This function will just generate the DOM node for the 404 graphic. Ideally you can have this static in the page with its display set to 'none' and then just keep flipping it 'block' and 'none'. But if you want all content creation from JS, then the above will work. Just generate and return the element.